src/stores/Binding/shared.ts

```diff
@@ -43,6 +43,33 @@ export const getEnabledCollectionNames = (
         .map(({ meta }) => meta.collectionName);
 };
 
+export const resetPartialCollectionMetadata = (
+    state: BindingState,
+    targetCollections?: string[]
+) => {
+    if (targetCollections) {
+        targetCollections.forEach((collection) => {
+            if (
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                state.collectionMetadata?.[collection] &&
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                !state.bindings?.[collection]
+            ) {
+                state.collectionMetadata[collection].added = false;
+                state.collectionMetadata[collection].sourceBackfillRecommended =
+                    false;
+            }
+        });
+
+        return;
+    }
+
+    Object.keys(state.collectionMetadata).forEach((collection) => {
+        state.collectionMetadata[collection].added = false;
+        state.collectionMetadata[collection].sourceBackfillRecommended = false;
+    });
```

Comment:
I kind of hate for even mentioning this... but I feel this is a spot where recursion might actually help. That way we don't have to keep the two things setting values in sync.